sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -465,6 +465,44 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
     }
   }
 
+  test("save csv with quote escaping enabled, using char to escape quote-escape") {
+    withTempDir { dir =>
+      val csvDir = new File(dir, "csv").getCanonicalPath
+
+
+      val df1 = spark.sqlContext.createDataFrame(List(
+        (1, """AA\"BB"""),              // 1 escape char anc 1 quote char
+        (2, """AA\\"BB"""),             // 2 escape char and 1 quote char
+        (3, """AA\"\"\"\"\"BB"""),      // (1 escape char anc 1 quote char) * 5
+        (4, """AA\\"\\"\\"\\"\\"BB"""), // (2 escape char and 1 quote char) * 5
+        (5, """You are "beautiful"""),
+        (6, """Yes, \"inside"\""")
+      ))
+
+      // escapeQuotes should be true by default
+      // escapeEscape should be equal to quote by default
+      df1.coalesce(1).write
+        .format("csv")
+        .option("quote", "\"")
+        .option("escape", "\\")
+        .option("escapeEscape", "\"")
+        .save(csvDir)
+
+      val df2 = spark.read
+        .format("csv")
+        .option("quote", "\"")
+        .option("escape", "\\")
+        .option("escapeEscape", "\"")
+        .load(csvDir).orderBy($"_c0")
```

Comment:
Per https://github.com/apache/spark/pull/17177/files#r105307099, I think we would not need this `orderBy`.